src/engine/IndexScan.cpp

```diff
@@ -306,18 +310,16 @@ ScanSpecificationAsTripleComponent IndexScan::getScanSpecificationTc() const {
 }
 
 // _____________________________________________________________________________
-ad_utility::HashMap<Variable, ColumnIndex>
-IndexScan::getVariableToSortedIndexMap() const {
-  ad_utility::HashMap<Variable, ColumnIndex> map;
-  ColumnIndex idx = 0;
-  for (const TripleComponent* const ptr : getPermutedTriple()) {
-    if (ptr->isVariable()) {
-      map[ptr->getVariable()] = idx;
-      break;
-    }
-    idx++;
+std::optional<std::pair<Variable, ColumnIndex>>
+IndexScan::getFirstSortedtVariableWithColumnIndex() const {
```

Comment:
This name has a typo, and also check my comment to the previous version of the code for good names and comments.